api.py

```diff
@@ -29,7 +29,55 @@ def get_algorithms():
 
 def _jsonify(response) -> Response:
     """Pass request data as a response"""
-    return flask.jsonify(response.json()), response.status_code
+    if response.ok:
+        return flask.jsonify(response.json()), response.status_code
+    return "{}", response.status_code
```

Comment:
How about informing the user what was the exact problem with the request?

Reply:
No - this API is public. We want to hide the errors, while developers have access to logs. It is usual approach for web servers due to security reasons.